### ISSUE TITLE
Clarifying install instructions in documentation

### ISSUE THREAD

**Issue by lchski** [wet-boew/wet-boew#7405]
Spent some time last night and today trying to get the repo up and running, and hit some walls when running `grunt`, which would fail due to an error related to `assemble`. (Specifically, `Warning: handlebars does not support render. Use --force to continue.`)

After some minor frustration, I had the idea to check the version of node specified in `.travis.yml`—`0.12`, compared to the `5.5.0` that I was running. So, I deleted `node_modules/` and `dist/` for the fourth or fifth time, used [nvm](https://github.com/creationix/nvm) to install the latest stable version of `0.12`, reran the install script, and had success!

The handlebars problem is related in some respect to the newer version of `node`/`npm` that I was running—reverting to `0.12` for this project solved that issue. (It wasn’t until afterwards that I found #6964 which would have helped me.) I think this could be better documented in [the install instructions](https://wet-boew.github.io/wet-boew/docs/start-en.html#develop), specifying explicitly which version of `node` is necessary to get the project running.

I’d be happy to take this on as a first contribution! Now that I’ve got the project up and running, I’d love to get involved with it.

**Comment by LaurentGoderre** [comment]
Hey, we definitely welcome any contribution. Don't worry about the French, I can take care of that.

However, ideally it would work in the latest version of npm

**Comment by lchski** [comment]
I definitely agree! However I think it has to do with some of the package versions we’re using. Searching the error that I ran into (`Warning: handlebars does not support render. Use --force to continue.`) returns quite a few results, with different symptoms/solutions:

* [Use the right handlebars dependency](https://github.com/assemble/assemble/issues/155)
* [npm versions 3+ use a flat `npm_modules/` structure, combined with version number strings being in the wrong format—an `npm-shrinkwrap` file helps](https://github.com/assemble/handlebars-helpers/issues/199)
* [`package.json` needs a version number *without dashes*](https://github.com/assemble/grunt-assemble/issues/31)

It looks like using only dots (no dashes) in the wet-boew `package.json`, and possibly switching the handlebars dependency, would solve the problem. I can take this on and experiment to see, though losing the dashed version in the `package.json` could be problematic, as there’d be no way to mark a development version.

**Comment by lchski** [comment]
I just did a quick test: removing the `-development` from the version string in `package.json` does indeed solve this problem. `grunt` ran without errors using `node 5.6.0` and `npm 3.6.0` on a fresh install of the repository. So that’s a quick fix, but it loses the ability to easily tag the codebase in the development state.

Alternatively, we could use an `npm-shrinkwrap.json` file to specify a version of whichever dependency is causing this error (one of the links above provides an example), but I haven’t tested that option yet.

**Comment by LaurentGoderre** [comment]
Huh??? How is that possible? The version number should not be affecting the Assemble tasks......

**Comment by lchski** [comment]
I have trouble understanding it either, but I assure you it worked when I ran it with a fresh install, using the latest node/npm, after removing `-development` from the `package.json` version string! [This comment](https://github.com/assemble/grunt-assemble/issues/31#issuecomment-158800396) explains the bizarre chain of events, while [this one](https://github.com/assemble/handlebars-helpers/issues/199#issuecomment-151965028) explains that swapping dots for dashes solves the problem. (But semver doesn’t allow for anything but a dash to delineate pre-release version strings, so we’re out of luck in that regard.)

**Comment by lchski** [comment]
Update:

I did some more testing, and realised that even a `package.json` version string with a `+` at the end creates issues: only decimals/dots work correctly when running the project with the latest version of node.

This is problematic, clearly—those version strings are pretty important, by my understanding. So, I tried the `npm-shrinkwrap.json` solution outlined [here](https://github.com/assemble/handlebars-helpers/issues/199#issuecomment-151970480), which worked, even with the dashes and plus signs in the version string! I’m putting the solution into a branch on my own fork so that you can see it.

**Comment by LaurentGoderre** [comment]
Yeah these are all valid semver. I wonder what causes different versions to be installed with different versions of npm

**Comment by lchski** [comment]
It’s not different versions being installed so much as it’s the `to` package misbehaving, by relying incorrectly on `handy` to determine :

**Comment by LaurentGoderre** [comment]
I would prefer updating assemble.

**Comment by lchski** [comment]
I’ll test out newer versions and, upon finding a known good one, make a PR!

**Comment by LaurentGoderre** [comment]
Awesome!!

**Comment by lchski** [comment]
Oof, this may be harder than it seemed. Takeaways/efforts so far:

* I tried to use [`assemble/grunt-assemble@^0.4.0`](https://github.com/assemble/grunt-assemble) instead of `assemble/assemble@^0.4.12`. Unfortunately, the former merely uses the latter with a few different wrappers/minor changes, which still produce the same errors in the end.
* Assemble post-0.4.42 (the version currently used by wet-boew) doesn’t work with grunt out-of-the-box... it relies on `grunt-assemble` to work, which hasn’t been updated to work with anything beyond 0.4.42.
* Assemble post-0.4.42 focuses on using `assemble` either as a standalone tool, or with gulp—grunt has been more or less shelved, though [the `grunt-assemble` team seems to be trying to update the assemble version it uses](https://github.com/assemble/grunt-assemble/issues/35).
* Updating to assemble post-0.4.42 would require significant retooling of our Gruntfile, which may involve even more inelegant hacks just to get it working.

***

I’m not sure what the best direction to head with this is—until either [`handy` and `to` are patched](https://github.com/assemble/handlebars-helpers/issues/199#issuecomment-151965028) (which seems very unlikely, given their more or less abandoned status) or `grunt-assemble` is updated to use a newer version of assemble.

* We could try to make the new version of assemble work with our grunt set up, but I have to confess that this is beyond me, as the new version of assemble works in very different ways.
* We could stop using dashes and plus signs in the wet-boew `package.json` version string. This is highly undesirable—it becomes very difficult to mark pre-release versions.
* We could use an `npm-shrinkwrap.json` file, following the [proof of concept](https://github.com/lchski/wet-boew/tree/fix-7405-npm-shrinkwrap) I worked up. The downside of this is that the `npm-shrinkwrap.json` would need to be updated each time a node module is added, removed, or a version number changed—with the nesting of `handy` under `to` repeated each time. (I *think* so, at least—from my understanding, that’s what we’d need to do.)

As far as I can tell, these are our options! Happy to keep plugging away at this, but not sure how much more helpful I can be.

**Comment by LaurentGoderre** [comment]
Wow! Thanks for all this research, saved me a lot of time.

Is there a way to just use shrinkwrap for a few dependencies? P.S. I am able to build with node v2.14.12 so perhaps it's only the latter version that are problematic. Maybe we can switch to the latest assemble when we go to the latest node.

**Comment by EricDunsworth** [comment]
@LaurentGoderre This may be of interest: npm/npm#7108

**Comment by lchski** [comment]
@EricDunsworth just beat me to it :) Definitely worth trying out.

As for switching to the latest assemble once going to the latest node, that’ll still require some retooling of the Gruntfile.

**Comment by hsrudnicki** [comment]
Does this remain an issue? If 'yes' then into consideration other issue related partially that is #7217
Please update the status of this, which is referenced to the 5574, whether this can be closed or otherwise, thank you

**Comment by hsrudnicki** [comment]
FYI: and there is a different issue: 6518 but of the same build category to be considered/included
